CommunityFridgeMapApi/dependencies/python/db.py

```diff
@@ -191,18 +191,60 @@ def add_item(self) -> DB_Response:
     def get_fridge_locations(self):
         pass
 
-
 class FrigeCheckIn(DB_Item):
-
     def __init__(self, db_client: 'botocore.client.DynamoDB'):
         super().__init__(db_client=db_client)
 
 class FridgeHistory(DB_Item):
-    
     def __init__(self, db_client: 'botocore.client.DynamoDB'):
         super().__init__(db_client=db_client)
 
+
 class Tag(DB_Item):
+    REQUIRED_FIELDS = ['tag_name']
+    TABLE_NAME = "tag"
 
-    def __init__(self, db_client: 'botocore.client.DynamoDB'):
+    def __init__(self, db_client: 'botocore.client.DynamoDB', tag_name:str= None):
         super().__init__(db_client=db_client)
+        self.tag_name = self.format_tag(tag_name)
+
+    def format_tag(self, tag_name:str) -> str:
+        #Tag_name is alphanumeric, can include hyphen and underscore, with no spaces and all lower cased
+        if tag_name:
+            tag_name = tag_name.lower().replace(" ", "")
+        return tag_name
+
+    def is_valid_tag_name(self, tag_name:str) -> bool:
+        #A valid tag name is alphanumeric, can include hyphen, underscore, with no spaces and all lower cased
+        if tag_name and len(tag_name) >= 3:
+            for x in tag_name:
+                if not x.isalnum() and x not in ['_', '-']:
+                    return False
+            return True
+        else:
+            return False
+
+    def add_item(self) -> DB_Response:
+        has_required_fields, field = self.has_required_fields()
+        if not has_required_fields:
+            return DB_Response(message = "Missing Required Field: %s" % field, status_code=400, success=False)
+        is_valid_field = self.is_valid_tag_name(self.tag_name)
+        if not is_valid_field:
+            return DB_Response(
+            message = "Tag Name Can Only Contain Letters, Numbers, Hyphens and Underscore: %s" % self.tag_name,
```

Comment:
We should probably set a max length too. What's a good max length for a tag do you think?

And also we should return a different error message if the length of the tag is the issue.  A good way to do this might be if `is_valid_tag_name` returns a tuple instead with a boolean and an error message. Where the error message is an empty string if it is a valid tag name. 

Side Note:

This is something that will probably be done in multiple classes so we could abstract it out. But probably not worth to worry about it right now. But something good to look up for fun would be how other projects do request validation on lambdas that use api gateway. I think api gateway has a form of request validation that takes the burden away from the developers, but I could be wrong

Reply:
An alternative to returning a tuple would be to raise an exception and catch it in the calling function